src/DevBetterWeb.Core/Interfaces/IDailyCheckPingService.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using DevBetterWeb.Core.Entities;
+
+namespace DevBetterWeb.Core.Interfaces
+{
+  public interface IDailyCheckPingService
+  {
+    Task SendPingIfNeeded(AppendOnlyStringList messages);
+    List<Invitation> CheckIfAnyActiveInvitationsRequireUserPing(List<Invitation> invitations);
+    List<Invitation> CheckIfAnyActiveInvitationsRequireAdminPing(List<Invitation> invitations);
+  }
```

Comment:
This is fine for now but has some obvious SRP and OCP violation issues.

A more extensible design would be to have a generic service that performs 0 to N checks, and another interface that defines a single check. The generic service would get an IEnumerable<CheckService> and loop over them, performing the checks. Then you could keep adding checks without having to add methods to an interface. Make sense?

Reply:
I see. Well, actually, the only reason these are public is so I can test them, because they're the main logic of the class. But I can have public methods on the implementation that aren't on the interface, right? Because they're called from SendPingIfNeeded

Reply:
Yes, you can certainly have public methods that aren't on the interface.